File: Aed-2/TPs/TP1/is.cpp

#include <stdio.h>
#include <ctype.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>


bool temVogal(char y)
{ // verifica se o char e vogal
// serve como funcao auxiliar ao metodo onlyvogais
  bool pa = false;
  if (y == 'a' || y == 'A' || y == 'e' || y == 'E' || y == 'i' || y == 'I' || y == 'o' || y == 'O' || y == 'u' || y == 'U')
  {pa = true;}
  return pa;
}
//Trata os caracteres especiais ("ignora eles e elimina algumas possibilidades de resposta")
bool eliminaEspecial(char cha[], int leng){
  bool pa = false;
  for (int j =0 ;j<leng;j = j + 1  ) {
          if((cha[j] >= '0' && cha[j] <= '9')){
           pa =true;
      }
  }
  return pa;
}

// verifica se o vetor de caracteres e composto apenas por consoantes
bool onlyVogais(char cha[], int leng)
{
  bool guarda = true;
  for (int j = 0; j < leng; j = j + 1)
  {
    if (!temVogal(cha[j]))// se aparecer qualquer coisa diferente de vogal sai do metodo
    {
      guarda = false;
    }
  }
  return guarda;
}
// verifica se o vetor de caracteres esta contido totalmente dentro desse intervalo(Letras)(Olha char a char)
bool contemletra(char y)
{
  bool pa = false;
  if (y >= 'A' && y <= 'Z' || y >= 'a' && y <= 'z')
    {pa = true;}
  return pa;
}

// verifica se o vetor de caracteres e composto apenas por consoantes
bool temConsoante(char cha[], int leng) 
{
  bool pa = true;
  for (int j = 0; j < leng; j = j + 1)
  {
    if (temVogal(cha[j]) || !contemletra(cha[j]))// se tiver vogal ou nao for letra sai do metodo
    {
      pa = false;
    }
  }
  return pa;
}

 // verifica se o vetor de caracteres e composto apenas por Numeros Inteiros
bool temInteiro(char cha[], int leng)
{
  bool pa = true;
  for (int j = 0; j < leng; j = j + 1)
  {
    if (contemletra(cha[j]) || cha[j] == ',' || cha[j] == '.') // exclui a possibilidade de letras e numeros Reais
    {pa = false;}
  }
  return pa;
} 

// verifica se o vetor de caracteres e composto apenas por Numeros Reais
bool temReal(char cha[], int leng)
{
  bool guarda = false;

  for (int i = 0; i < leng; i++)
  {
    if (!contemletra(cha[i])) // exclui a possibilidade de existerem letrar 
    {
      guarda = eliminaEspecial(cha,leng);
    }
  }
  return guarda;
}


    

int main() // metodo principal
{

  char palavra[3000];
  
  fgets(palavra, 400, stdin);
  palavra[strlen(palavra) - 1] = '\0';
  int leng = strlen(palavra) - 1;

 bool pa = true;
  while (pa == true)
  {
   // verifica se a entrada  ==  flag (saida)
  if (strlen(palavra) == 3 && palavra[0] == 'F' && palavra[1] == 'I' && palavra[2] == 'M') 
       {
        pa = false;
       }
else{
    if ( onlyVogais(palavra, leng) == true)
    { printf("SIM ");}
    else{printf("NAO ");}

    if (temConsoante(palavra, leng) == true)
    {  printf("SIM ");}
    else
    {  printf("NAO ");}

    if (temInteiro(palavra, leng) == true)
    { printf("SIM ");}
    else
    {  printf("NAO ");}

    if (temReal(palavra, leng) == true)
    {  printf("SIM\n");}
    else
    {  printf("NAO\n");}

    
    fgets(palavra, 400, stdin); // le novamente para reentrar na verificacao da palavra
    palavra[strlen(palavra) - 1] = '\0';
    leng = strlen(palavra) - 1;
}
  }
}

File: Aed-2/TPs/TP1/is.java
public class is {
  public static void main(String[] args) { // funcao main (printa os resultados processados nas funcoes)
    String palavra = MyIO.readLine();
    while (Vef(palavra) == true) {
      if (vogal(palavra) == true) {
        System.out.print("SIM ");
      } else {
       System.out.print("NAO ");
      }
      if (consoante(palavra) == true) {
        System.out.print("SIM ");
      } else {
        System.out.print("NAO ");
      }
      if (temNumeroInteiro(palavra) == true) {
        System.out.print("SIM ");
      } else {
        System.out.print("NAO ");
      }
      if (temNumeroReal(palavra) == true) {
        System.out.print("SIM\n");
      } else {
        System.out.print("NAO\n");
      }

      palavra = MyIO.readLine();
    }

  }

  public static boolean vogal(String y) { // verifica se sao so vogais
    char x;
    boolean trader = true;
    for (int i = 0; i < y.length() - 1; i = i + 1) {
      x = y.charAt(i);

      if( !(x == 'a' || x == 'A' || x == 'e' || x == 'E' || x == 'i' || x == 'u' || x == 'U' || x == 'o' || x == 'O')) {
        trader = false;
       
      }
  
    }
    return trader;
  }

  public static boolean consoante(String y) { //verigica se sao so consoantes
    char x;
    boolean trader = true;
    for (int i = 0; i < y.length() - 1; i = i + 1) {
      x = y.charAt(i);

      if ((x == 'a' || x == 'A' || x == 'e' || x == 'E' || x == 'i' || x == 'u' || x == 'U' || x == 'o' || x == 'O') || !dentro(y.charAt(i))) {
        trader = false;
       
      }
  
    }
    return trader;
  }

 public static boolean desaparecaEspecial(String y) // retira caracteres indesejados
 {
   boolean ret1 = false;
   for(int i = 0; i<y.length();i = i + 1)
   {
     if((y.charAt(i) >= '0' && y.charAt(i) <= '9'))
     ret1 = true;
   }
   return ret1;
 }


 public static boolean dentro(char x) // verifica se é letra
 {
   boolean ret1 = false;
if(x >= 'A' && x <= 'Z' || x >= 'a' && x <= 'z')
ret1 = true;

   return ret1;
 }

 public static boolean temNumeroInteiro (String y){ // verifica se a string so tem numeros inteiros
  boolean saber = true;
   for (int i =0 ;i<y.length();i++  ) {

           if(y.charAt(i)=='.'||y.charAt(i)==','||dentro(y.charAt(i))){

             saber = false;

           }
     }
 return saber; 
}
public static boolean temNumeroReal(String y){ //verifica se so tem numeros reais

  boolean saber = false;
 // System.out.println(pato);
   for (int i =0 ;i<y.length();i++  ) {

           if(!dentro(y.charAt(i)) ){

             saber = desaparecaEspecial(y);
           }
     }
 return saber;

}



  public static boolean Vef(String teste) {// Metodo para verificar o flag
    boolean pa = true;

    if (teste.length() == 3 && teste.charAt(0) == 'F' && teste.charAt(1) == 'I' && teste.charAt(2) == 'M') {
      pa = false;
    }
    return pa;
  }

}